### PR TITLE
Adding more actions

### PR DESCRIPTION
**Pull request by vanajamukkara**

### CODE REVIEW

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java [hunk 1]
```diff
@@ -14,30 +21,225 @@
...
     public static boolean installOperator() {
         return Operator.install();
     }
 
+    /**
+     * Upgrade existing Operator release
+     * @param values map with values
+     * @return true if the operator is successfully upgraded, false otherwise.
+     */
+    public static boolean upgradeOperator(HashMap<String, String> values) {
```
Comment (rjeberhard) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java]: We should make a fluent API for the helm chart values

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java [hunk 2]
```diff
@@ -14,30 +21,225 @@
...
     // ----------------------   domain  -----------------------------------
 
+    /**
+     * Create domain custom resource from the given domain yaml file.
+     * @param domainUID
+     * @param namespace
+     * @param domainYAML
+     * @return true on success, false otherwise
+     */
     public static boolean createDomainCustomResource(String domainUID, String namespace, String domainYAML) {
```
Comment (rjeberhard) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java]: Fluent api for domain... I'm willing to help with these :)

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java [hunk 3]
```diff
@@ -14,30 +21,225 @@
...
+    }
+
     // ------------------------   ingress controller ----------------------
 
+    /**
+     * Install Traefik Operator
+     * @param valuesYaml values yaml file to be used
+     * @return true on success, false otherwise
+     */
+    public static boolean installTraefik(String valuesYaml) {
```
Comment (rjeberhard) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java]: Similarly, let's make the tests more expressive.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java [hunk 4]
```diff
@@ -14,30 +21,225 @@
...
     }
 
+    // -------------------------   pv/pvc  ---------------------------------
+
+    /**
+     * Create Kubernetes Persistent Volume using the yaml provided
+     * @param pvYaml the persistent volume yaml file
+     * @return true on success, false otherwise
+     */
+    public static boolean createPersistentVolume(String pvYaml) {
```
Comment (rjeberhard) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java]: We have an API for standard types. `io.kubernetes.client.openapi.models.V1PersistentVolume`

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java [hunk 5]
```diff
@@ -14,30 +21,225 @@
...
+    public static boolean deletePersistentVolume(String pvName) {
+        return PersistentVolume.delete(pvName);
+    }
+
+    /**
+     * Create Kubernetes Persistent Volume Claim using the yaml provided
+     * @param pvcYaml the persistent volume claim yaml file
+     * @return true on success, false otherwise
+     */
+    public static boolean createPersistentVolumeClaim(String pvcYaml) {
```
Comment (rjeberhard) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java]: Similarly, have existing type

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java (hunk 1 above)
Comment (markxnelson) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java]: agree

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java (hunk 4 above)
Comment (markxnelson) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java]: happy to discuss, but if it is ugly, i would prefer to have a fluent api that hides that ugliness under the covers

Comment (rjeberhard) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java]: These types already provide a fluent API: https://github.com/kubernetes-client/java/blob/master/examples/src/main/java/io/kubernetes/client/examples/FluentExample.java

Comment (markxnelson) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java]: prefect :)

Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java]: Shall I remove the action methods for all standard types?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java (hunk 1 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java]: work on fluent API for operator, domain and update the PR?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java [hunk 6]
```diff
@@ -7,31 +7,86 @@
...
+    public boolean delete(){
         return true;
     }
 
-    public static boolean deleteRelease(String name) {
+    public boolean addRepo() {
         return true;
     }
 
+    public static class HelmBuilder {
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java]: do we need the builder? i did a lot of googling and i did not come up with a definitive answer. this seems to be builder pattern, which is almost the same as fluent, but not exactly. i think fluent is more of a static that manages the instance. i don't mind really - i will defer to @rjeberhard and @ddsharpe on this

Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java]: fluent example https://github.com/kubernetes-client/java/blob/master/examples/src/main/java/io/kubernetes/client/examples/FluentExample.java @rjeberhard suggested uses builder, thought we wanted to use that pattern. Not sure if he meant to use that only for Operator or Domain.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java [hunk 7]
```diff
@@ -28,8 +29,8 @@
      * @param values map with values
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java]: fix javadoc

**Pull request merged**